install/install_requirements.sh

```diff
@@ -9,26 +9,39 @@ set -eou pipefail
 
 # Install required python dependencies for developing
 # Dependencies are defined in .pyproject.toml
-PYTHON_EXECUTABLE=${PYTHON_EXECUTABLE:-python}
-if [[ -z ${CONDA_DEFAULT_ENV:-} ]] || [[ ${CONDA_DEFAULT_ENV:-} == "base" ]] || [[ ! -x "$(command -v python)" ]];
+if [ -z "${PYTHON_EXECUTABLE:-}" ];
 then
-  PYTHON_EXECUTABLE=python3
+  if [[ -z ${CONDA_DEFAULT_ENV:-} ]] || [[ ${CONDA_DEFAULT_ENV:-} == "base" ]] || [[ ! -x "$(command -v python)" ]];
+  then
+    PYTHON_EXECUTABLE=python3
+  fi
 fi
+echo "Using python executable: $PYTHON_EXECUTABLE"
 
+PYTHON_SYS_VERSION="$($PYTHON_EXECUTABLE -c "import sys; print(f'{sys.version_info.major}.{sys.version_info.minor}')")"
 # Check python version. Expect 3.10.x or 3.11.x
-printf "import sys\nif sys.version_info.major != 3 or sys.version_info.minor < 10 :\n\tprint('Please use Python >=3.10');sys.exit(1)\n" | $PYTHON_EXECUTABLE
-if [[ $? -ne 0 ]]
+if ! $PYTHON_EXECUTABLE -c "
+import sys
+if sys.version_info < (3, 10) or sys.version_info >= (3, 12):
```

Comment:
what's wrong with 3.12? that seems to be what I have on my Mac and I don't think I've updated recently.

Reply:
It was based on the existing comments (which we just happen to not enforce)

That said, let's open up 3.12 since that's been looking fine to me

Reply:
That's what I get for reviewing code after 1AM

Reply:
I don’t see any issues with 3.12. The main question is less about whether it works on 3.12 and more about which version the project currently aims to support. Opened a followup [PR](https://github.com/pytorch/torchchat/pull/1370) to address this.